modules/imgcodecs/test/test_read_write.cpp

```diff
@@ -520,6 +520,22 @@ TEST(ImgCodecs, multipage_collection_two_iterator_operatorpp)
          EXPECT_TRUE(cv::norm(img1, img[i], NORM_INF) == 0);
     }
 }
+
+// See https://github.com/opencv/opencv/issues/26207
+TEST(Imgcodecs, imencodemulti_regression_26207)
+{
+    vector<Mat> imgs;
+    Mat img(100, 100, CV_8UC1);
+    imgs.push_back(img);
```

Comment:
@Kumataro could you test 
```
   vector<Mat> imgs;
    Mat img(100, 100, CV_8UC1);
    imgs.push_back(img);
    imgs.push_back(img.clone());
```

Reply:
Thank you for your comment ! I mistakenly thought that it was enough to support multiple images for imencode().

But it seems that encoding multiple images requests reading callback function. After appending it, it works well.

Reply:
> Thank you for your comment ! I mistakenly thought that it was enough to support multiple images for imencode().
> 
> But it seems that encoding multiple images requests reading callback function. After appending it, it works well.

Thank you for your contribution. to me looks perfect.